envoy-control-core/src/test/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactoryTest.kt

```diff
@@ -0,0 +1,71 @@
+import pl.allegro.tech.servicemesh.envoycontrol.groups.IncomingEndpoint
+import pl.allegro.tech.servicemesh.envoycontrol.groups.PathMatchingType
+import pl.allegro.tech.servicemesh.envoycontrol.snapshot.IncomingPermissionsProperties
+import org.assertj.core.api.Assertions.assertThat
+
+internal class RBACFilterFactoryTest {
+    val rbacFilterFactory = RBACFilterFactory(IncomingPermissionsProperties())
+
+    val expected = RBAC.newBuilder().putAllPolicies(mapOf(
+            "client1,client2" to Policy.newBuilder().addAllPrincipals(listOf(
+                    Principal.newBuilder().setHeader(
+                            HeaderMatcher.newBuilder().setName("x-service-name").setExactMatch("client1").build()
+                    ).build(),
+                    Principal.newBuilder().setHeader(
+                            HeaderMatcher.newBuilder().setName("x-service-name").setExactMatch("client2").build()
+                    ).build()
+            )).addAllPermissions(listOf(
+                Permission.newBuilder().setAndRules(
+                    Permission.Set.newBuilder().addAllRules(
+                            listOf(
+                                    Permission.newBuilder().setHeader(
+                                            HeaderMatcher.newBuilder()
+                                                    .setName(":path").setExactMatch("/example").build()
+                                    ).build(),
+                                Permission.newBuilder().setOrRules(
+                                        Permission.Set.newBuilder().addAllRules(
+                                                listOf(
+                                                    Permission.newBuilder().setHeader(
+                                                            HeaderMatcher.newBuilder()
+                                                                    .setName(":method").setExactMatch("GET").build()
+                                                    ).build(), Permission.newBuilder().setHeader(
+                                                            HeaderMatcher.newBuilder()
+                                                                    .setName(":method").setExactMatch("POST").build()
+                                                    ).build()
+                                                )
+                                        ).build()
+                                ).build()
+                            )
+                    )
+                ).build()
+            )).build()
+    )).build()
+
+    @Test
+    fun `should generate RBAC rules for simple incoming permissions`() {
+        // given
+        val incomingPermission = Incoming(
+                endpoints = listOf(IncomingEndpoint(
+                        "/example",
+                        PathMatchingType.PATH,
+                        setOf("GET", "POST"),
+                        setOf("client1", "client2")
```

Comment:
In current incoming permissions test we have a test case with 2 endpoint. IMHO it was already a limted test, I would not reduce it even more :)


I think it would be better to not reduce incoming permissions test cases even more.